### ISSUE TITLE
Circular build depencency for developers because of y.tab.h

### ISSUE THREAD

**Issue by GoogleCodeExporter** [wainstead/waverous#18]
```
From the generated Makefile, I wrote:

# You have to run "make," thus compiling the project first before you
# can run "make depend." This is because "y.tab.h" is generated by flex
# during the compilation process:
#   In file included from keywords.c:29:
#   tokens.h:23:19: error: y.tab.h: No such file or directory
# Without y.tab.h "make depend" fails with this error.

The solution to this problem is redo the way 'make depend' works, perhaps as 
simply as adding more lines to the shell commands to rectify the problem.

In other words, the goal here is to get rid of having to run "make" in order to 
run "make depend."

It's possible that further development of the autotools files in this project 
-- getting to the point of using automake and autoconf together -- might render 
this ticket obsolete.
```

Original issue reported on code.google.com by `wainst...@gmail.com` on 15 Jan 2012 at 5:15
